core/common/src/utils.rs

```diff
@@ -171,7 +179,35 @@ pub async fn get_service_uri(
         }
     };
 
-    Ok(result)
+    let uri = get_uri(&result).map_err(|err| {
+        Status::failed_precondition(format!(
+            "Unable to get container environment variable with error: {err}"
+        ))
+    })?;
+
+    Ok(uri)
+}
+
+/// If feature 'containerize' is set, will modify a localhost uri to point to container's localhost
+/// DNS alias. Otherwise, returns the uri as a String.
+///
+/// # Arguments
+/// * `uri` - The uri to potentially modify.
+pub fn get_uri(uri: &str) -> Result<String, Box<dyn std::error::Error + Send + Sync>> {
+    #[cfg(feature = "containerize")]
+    let uri = {
+        // Container env variable names.
+        const HOST_GATEWAY_ENV_VAR: &str = "HOST_GATEWAY";
```

Comment:
Perhaps these constants should be moved out of this function (perhaps to s shared constant code file) and made public?

Reply:
So the idea was to keep the constants with this function because it is only used in a container, and those values should be set by the .env files provided

Reply:
If they are only ever used once, then they probably don't even need to be constants.